jaraco/develop/add-github-secret.py

```diff
@@ -1,8 +1,16 @@
-import autocommand
+import typer
+from typing_extensions import Annotated
+from jaraco.ui.main import main
 
 from . import github
 
 
-@autocommand.autocommand(__name__)
-def run(name, value, project: github.Repo = github.Repo.detect()):
+@main
+def run(
+    name,
+    value,
+    project: Annotated[
+        github.Repo, typer.Option(parser=github.Repo)
+    ] = github.Repo.detect(),
```

Comment:
Because `github.Repo` is a subclass of `str`, it can be constructed from a simple string, but that string needs to be the name of the project in GitHub. I should probably document that.

I think what you're suggesting is that `github.Repo.detect(...)` should accept a `path` parameter that defaults to `.` instead of assuming the repo is in the `cwd`. I'd be okay with that, I think.

Regardless, this change is about something different - migrating from autocommand to typer, so I'm aiming for feature parity. We can address other concerns in another issue/PR.